styles/snapchat-web/catppuccin.user.css

```diff
@@ -0,0 +1,116 @@
+    @mocha:     { @rosewater: #f5e0dc; @flamingo: #f2cdcd; @pink: #f5c2e7; @mauve: #cba6f7; @red: #f38ba8; @maroon: #eba0ac; @peach: #fab387; @yellow: #f9e2af; @green: #a6e3a1; @teal: #94e2d5; @sky: #89dceb; @sapphire: #74c7ec; @blue: #89b4fa; @lavender: #b4befe; @text: #cdd6f4; @subtext1: #bac2de; @subtext0: #a6adc8; @overlay2: #9399b2; @overlay1: #7f849c; @overlay0: #6c7086; @surface2: #585b70; @surface1: #45475a; @surface0: #313244; @base: #1e1e2e; @mantle: #181825; @crust: #11111b; };
+  }
+
+  #catppuccin(@lookup, @accent) {
+    @rosewater: @catppuccin[@@lookup][@rosewater];
+    @flamingo: @catppuccin[@@lookup][@flamingo];
+    @pink: @catppuccin[@@lookup][@pink];
+    @mauve: @catppuccin[@@lookup][@mauve];
+    @red: @catppuccin[@@lookup][@red];
+    @maroon: @catppuccin[@@lookup][@maroon];
+    @peach: @catppuccin[@@lookup][@peach];
+    @yellow: @catppuccin[@@lookup][@yellow];
+    @green: @catppuccin[@@lookup][@green];
+    @teal: @catppuccin[@@lookup][@teal];
+    @sky: @catppuccin[@@lookup][@sky];
+    @sapphire: @catppuccin[@@lookup][@sapphire];
+    @blue: @catppuccin[@@lookup][@blue];
+    @lavender: @catppuccin[@@lookup][@lavender];
+    @text: @catppuccin[@@lookup][@text];
+    @subtext1: @catppuccin[@@lookup][@subtext1];
+    @subtext0: @catppuccin[@@lookup][@subtext0];
+    @overlay2: @catppuccin[@@lookup][@overlay2];
+    @overlay1: @catppuccin[@@lookup][@overlay1];
+    @overlay0: @catppuccin[@@lookup][@overlay0];
+    @surface2: @catppuccin[@@lookup][@surface2];
+    @surface1: @catppuccin[@@lookup][@surface1];
+    @surface0: @catppuccin[@@lookup][@surface0];
+    @base: @catppuccin[@@lookup][@base];
+    @mantle: @catppuccin[@@lookup][@mantle];
+    @crust: @catppuccin[@@lookup][@crust];
+    @accent-color: @catppuccin[@@lookup][@@accent];
+
+    /* Snapchat applies the variables (overrides?) to specific elements so we need to apply to every element (with high specificity) to override those. */
+    &,
+    body > main#root * {
+      --sigMain: @base;
+      --sigSurface: @mantle;
+      --sigSurfaceRGB: red(@mantle) green(@mantle) blue(@mantle);
+      --sigAboveSurface: @crust;
+      --sigSurfaceDown: @base;
+      --sigSubscreen: @crust;
+      --sigTextField: @crust;
+      --sigDivider: @surface0;
+      --sigDividerLight: @surface0;
+      --sigBackgroundPrimary: @base;
+      --sigBackgroundSecondary: @mantle;
+      --sigBackgroundSecondaryHover: darken(@mantle, 5%);
+      --sigBackgroundTertiary: @crust;
+      --sigBackgroundMessageHover: lighten(@crust, 7%);
+      --sigOverlay: fade(@overlay0, 0.4);
+      --sigOverlayHover: fade(@overlay0, 0.35);
+
+      --sigTextPrimary: @text;
+      --sigButtonOnPrimary: @text;
```

Comment:
```suggestion
      --sigButtonOnPrimary: @crust;
```

This should fix the poor contrast here 
![image](https://github.com/catppuccin/userstyles/assets/71222764/ab963254-8b24-42af-a07b-1907eda2e2ef)